exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlConverter.java

```diff
@@ -250,20 +250,22 @@ protected void validateFrom(
         RelDataType targetRowType,
         SqlValidatorScope scope) {
       switch (node.getKind()) {
-      case AS:
-        if (((SqlCall) node).operand(0) instanceof SqlIdentifier) {
-          SqlIdentifier tempNode = ((SqlCall) node).operand(0);
-          DrillCalciteCatalogReader catalogReader = (SqlConverter.DrillCalciteCatalogReader) getCatalogReader();
-
-          // Check the schema and throw a valid SchemaNotFound exception instead of TableNotFound exception.
-          if (catalogReader.getTable(Lists.newArrayList(tempNode.names)) == null) {
-            catalogReader.isValidSchema(tempNode.names);
+        case AS:
+          if (((SqlCall) node).operand(0) instanceof SqlIdentifier) {
+            SqlIdentifier tempNode = ((SqlCall) node).operand(0);
+            DrillCalciteCatalogReader catalogReader = (SqlConverter.DrillCalciteCatalogReader) getCatalogReader();
+
+            // Check the schema and throw a valid SchemaNotFound exception instead of TableNotFound exception.
+            if (catalogReader.getTable(Lists.newArrayList(tempNode.names)) == null) {
+              catalogReader.isValidSchema(tempNode.names);
+            }
+            changeNamesIfTableIsTemporary(tempNode);
           }
-          changeNamesIfTableIsTemporary(tempNode);
-        }
-      default:
-        super.validateFrom(node, targetRowType, scope);
+          break;
+        default:
+          break;
```

Comment:
doesn't look like this change is needed. It's not changing any behavior ?

Reply:
It is not changing behavior. But this way we could add more Kind(s) here e.g. UNNEST.